zerver/lib/actions.py

```diff
@@ -4578,6 +4579,47 @@ def do_update_message(user_profile: UserProfile, message: Message, topic_name: O
         else:
             event['wildcard_mention_user_ids'] = []
 
+    if new_stream is not None:
+        # For simplicity, right now we only allow changing the stream,
+        # not also the topic at the same time.
+        assert topic_name is None
```

Comment:
I'm now kinda tempted to just add backend support for changing `topic_name` at the same time; having seen this work, I think it's probably about the same amount of code (mostly, we need to change the outcome of the relevant test to verify the messages ended up at the new topic).  Can you take care of that?

Reply:
Sure, let me see.